azkaban-common/src/main/java/azkaban/imagemgmt/daos/ImageTypeDaoImpl.java

```diff
@@ -44,31 +48,34 @@
   private final DatabaseOperator databaseOperator;
 
   static String INSERT_IMAGE_TYPE =
-      "insert into image_types ( name, description, active, deployable, created_by, modified_by )"
-          + " values (?, ?, ?, ?, ?, ?)";
+      "insert into image_types ( name, description, active, deployable, created_by, "
+          + "created_on, modified_by, modified_on )"
+          + " values (?, ?, ?, ?, ?, ?, ?, ?)";
   static String INSERT_IMAGE_OWNERSHIP =
-      "insert into image_ownerships ( type_id, owner, role, created_by, modified_by ) "
-          + "values (?, ?, ?, ?, ?)";
+      "insert into image_ownerships ( type_id, owner, role, created_by, created_on, modified_by, "
+          + "modified_on ) values (?, ?, ?, ?, ?, ?, ?)";
 
   @Inject
-  public ImageTypeDaoImpl(DatabaseOperator databaseOperator) {
+  public ImageTypeDaoImpl(final DatabaseOperator databaseOperator) {
     this.databaseOperator = databaseOperator;
   }
 
   @Override
-  public int createImageType(ImageType imageType) {
+  public int createImageType(final ImageType imageType) {
     final SQLTransaction<Integer> insertAndGetSpaceId = transOperator -> {
       // insert image type record
-      transOperator.update(INSERT_IMAGE_TYPE, imageType.getName(), imageType.getDescription(),
-          true, imageType.getDeployable().getName(), imageType.getCreatedBy(),
-          imageType.getModifiedBy());
-      int imageTypeId = Long.valueOf(transOperator.getLastInsertId()).intValue();
+      // Passing timestamp from the code base and can be formatted accordingly based on timezone
+      final Timestamp currentTimestamp = Timestamp.valueOf(LocalDateTime.now());
+      transOperator.update(INSERT_IMAGE_TYPE, imageType.getName().toLowerCase(),
+          imageType.getDescription(), true, imageType.getDeployable().getName(),
+          imageType.getCreatedBy(), currentTimestamp, imageType.getModifiedBy(), currentTimestamp);
```

Comment:
This change is purposefully done. Inserting timestamp such as created and updated from the codebase as we will have more control and timestamp formatting can be done based on timezone. There is also another reason for these changes as the custom code which is invoked as part of Azkaban DatabaseSetup unable to process some of the create table definition containing on update clause. So, it make sense to update the timestamp from one place that is codebase. Hence, in order maintain consistency and  correct timezone, timestamps are inserted from codebase only.

Reply:
The MySQL TIMESTAMP data type automatically converts all date-time values to UTC, so even if we use the local time zone when assigning creation/modification times in the code (which I strongly suggest NOT to do), it will be saved in UTC. When retrieving TIMESTAMP values MySQL convert them to local time.

Reply:
The MySQL TIMESTAMP data type automatically converts all date-time values to UTC, so even if we use the local time zone when assigning creation/modification times in the code (which I strongly suggest NOT to do), it will be saved in UTC. When retrieving TIMESTAMP values MySQL convert them to local time.

Reply:
I would love to keep timestamp insertion at the db level. As mentioned in my earlier reply during update it has to be inserted from the codebase. That will be make things inconsistent. In future we can definitely keep timestamp insertion at the db level if address our current limitation.